main.js

```diff
@@ -871,6 +1049,7 @@ function updateSaveValuesFromForm(values) {
     values[427] = document.getElementById("sav-muffetstate").value;
     values[432] = document.getElementById("sav-broguardsstate").value;
     values[455] = document.getElementById("sav-mettatonstate").value;
+    values[495] = +document.getElementById("sav-undyne-cell").checked;
```

Comment:
This should probably be more explicit (in other places, too):
```suggestion
    values[495] = Number(document.getElementById("sav-undyne-cell").checked);
```